Formula/cvs-fast-export.rb

```diff
@@ -0,0 +1,23 @@
+class CvsFastExport < Formula
+  desc "Export an RCS or CVS history as a fast-import stream"
+  homepage "http://www.catb.org/~esr/cvs-fast-export/"
+  url "http://www.catb.org/~esr/cvs-fast-export/cvs-fast-export-1.55.tar.gz"
+  sha256 "af58e16667c6e02e8431ff666effe306d3b30086fab483170257890da1afc21b"
+  license "GPL-2.0-or-later"
+
+  depends_on "asciidoc" => :build
+  depends_on "docbook" => :build
+
+  def install
+    # otherwise asciidoc will fail to find docbook
+    ENV["XML_CATALOG_FILES"] = etc/"xml/catalog"
+
+    # warning: make takes an unusually long time to complete
+    system "make", "cvs-fast-export", "man"
+    system "make", "install", "prefix=#{prefix}"
+  end
+
+  test do
+    assert_match "cvs-fast-export: version #{version}", shell_output("#{bin}/cvs-fast-export --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev - `cvs-fast-export` relies on a CVS repository as input, so I'm having a hard time coming up with a simple test case. Is it preferred to write a test that requires `cvs` be installed to create a simple repo?

Reply:
OK I've added a full test using cvs, thanks for the help! Can you please review @SMillerDev ?